## The Wind data analysis
Here we download the dataset which contains the wind speed in Uzbekistan. The reason why I have selected Uzbekistan is that it is stepping toward green economy, and renewable energy is the part of Green Economy. So, I am going for wind analysis in Uzbekistan and whether it is efficient to implement wind turbine in my country.

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv(r"C:\Users\sardo\OneDrive\Desktop\weather archive from 2020 to 2023.csv")

In [3]:
df

time  windspeed_10m (km/h)  windspeed_100m (km/h)  \
0      1/1/2020                  11.7                   21.9   
1      1/1/2020                  12.0                   21.8   
2      1/1/2020                  12.0                   21.9   
3      1/1/2020                  12.5                   23.0   
4      1/1/2020                  13.0                   24.0   
...         ...                   ...                    ...   
30715  7/3/2023                  16.1                   27.3   
30716  7/3/2023                  15.3                   27.1   
30717  7/3/2023                  15.4                   27.9   
30718  7/3/2023                  13.7                   25.9   
30719  7/3/2023                  13.0                   25.2   

       windgusts_10m (km/h)  
0                      16.9  
1                      16.9  
2                      16.9  
3                      17.6  
4                      18.7  
...                     ...  
30715                  23.4  
30716                  22.3  
30717                  22.3  
30718                  22.3  
30719                  19.4  

[30720 rows x 4 columns]

As it is shown in the above table, there are 4 columns and 30720 rows. Our data set was recorded hourly.
Note! windspeed_10m and windspeed_100m mean wind speed at 10 and 100 meters above ground. Wind speed on 10 meters is the standard level.

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30720 entries, 0 to 30719
Data columns (total 4 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   time                   30720 non-null  object 
 1   windspeed_10m (km/h)   30720 non-null  float64
 2   windspeed_100m (km/h)  30720 non-null  float64
 3   windgusts_10m (km/h)   30720 non-null  float64
dtypes: float64(3), object(1)
memory usage: 960.1+ KB


In [5]:
df.describe()

windspeed_10m (km/h)  windspeed_100m (km/h)  windgusts_10m (km/h)
count          30720.000000           30720.000000          30720.000000
mean              15.016631              22.014502             24.306348
std                6.851886              10.035571             10.776566
min                0.000000               0.000000              1.100000
25%               10.100000              14.400000             16.200000
50%               14.200000              21.900000             23.000000
75%               19.200000              28.800000             31.000000
max               52.100000              69.100000             79.600000

Now we have got overview of the dataset. We need the average, min, max speed of wind. According to the information of the eia.gov, good places for wind turbines are where the annual average wind speed is at least 9 miles per hour(mph) or 14 kilometer per hour(km/h) for small wind turbines and 13 mph (20 km/h) for utility-scale turbines.

Now we have hourly, daily, monthly data about wind in Uzbekistan. It will be more easy and understandable if we change this info into daily data. It is easy to use underscores instead of brackets.

In [6]:
df2 = df.set_axis(['time', 'windspeed_10m_kmh', 'windspeed_100m_kmh', 'windgusts_10m_kmh'], axis = 'columns')
df2

time  windspeed_10m_kmh  windspeed_100m_kmh  windgusts_10m_kmh
0      1/1/2020               11.7                21.9               16.9
1      1/1/2020               12.0                21.8               16.9
2      1/1/2020               12.0                21.9               16.9
3      1/1/2020               12.5                23.0               17.6
4      1/1/2020               13.0                24.0               18.7
...         ...                ...                 ...                ...
30715  7/3/2023               16.1                27.3               23.4
30716  7/3/2023               15.3                27.1               22.3
30717  7/3/2023               15.4                27.9               22.3
30718  7/3/2023               13.7                25.9               22.3
30719  7/3/2023               13.0                25.2               19.4

[30720 rows x 4 columns]

In [7]:
daily_average = df2.groupby(['time'], sort = False).mean()
daily_min = df2.groupby(['time'], sort = False).min()
daily_max = df2.groupby(['time'], sort = False).max()

In [8]:
daily_average

windspeed_10m_kmh  windspeed_100m_kmh  windgusts_10m_kmh
time                                                               
1/1/2020           11.812500           18.629167          18.266667
1/2/2020            5.137500            6.145833           8.341667
1/3/2020            9.141667           13.287500          14.570833
1/4/2020           12.162500           16.725000          20.012500
1/5/2020           12.866667           19.366667          19.533333
...                      ...                 ...                ...
6/29/2023          17.333333           24.979167          28.387500
6/30/2023          15.283333           22.379167          26.033333
7/1/2023           12.875000           17.791667          21.916667
7/2/2023           18.608333           23.875000          29.279167
7/3/2023           12.870833           18.462500          22.670833

[1280 rows x 3 columns]

In [9]:
daily_min

windspeed_10m_kmh  windspeed_100m_kmh  windgusts_10m_kmh
time                                                               
1/1/2020                 7.5                 9.7               11.2
1/2/2020                 0.5                 1.0                2.5
1/3/2020                 2.3                 5.4                6.8
1/4/2020                 4.5                 6.9                6.8
1/5/2020                 7.4                10.8               11.9
...                      ...                 ...                ...
6/29/2023               14.7                17.6               20.5
6/30/2023               13.2                15.9               19.8
7/1/2023                 1.5                 5.4                6.1
7/2/2023                 5.9                 5.4                9.7
7/3/2023                 2.3                 6.8                5.4

[1280 rows x 3 columns]

In [10]:
daily_max

windspeed_10m_kmh  windspeed_100m_kmh  windgusts_10m_kmh
time                                                               
1/1/2020                15.3                24.9               24.8
1/2/2020                 9.3                15.2               11.2
1/3/2020                15.0                27.3               28.4
1/4/2020                18.4                26.7               29.5
1/5/2020                16.4                26.7               27.0
...                      ...                 ...                ...
6/29/2023               20.2                34.8               33.1
6/30/2023               16.9                31.1               32.4
7/1/2023                20.6                31.4               37.1
7/2/2023                35.2                42.1               47.5
7/3/2023                16.6                27.9               30.2

[1280 rows x 3 columns]

Now we have sorted the data into daily average, min and max.
Next it is more helpful how many days wind blows faster than 9 mph or 14 km/h because this is the minimum speed at which wind turbines can generate electricity.

In [11]:
daily_average[(daily_average['windspeed_10m_kmh'] > 14) & (daily_average['windspeed_100m_kmh'] > 14)]

windspeed_10m_kmh  windspeed_100m_kmh  windgusts_10m_kmh
time                                                               
1/8/2020           16.675000           26.991667          24.275000
1/9/2020           21.462500           34.208333          32.341667
1/14/2020          15.525000           23.533333          22.879167
1/22/2020          20.654167           31.166667          31.987500
1/23/2020          16.170833           26.500000          26.220833
...                      ...                 ...                ...
6/27/2023          15.158333           20.904167          24.912500
6/28/2023          18.333333           25.587500          29.904167
6/29/2023          17.333333           24.979167          28.387500
6/30/2023          15.283333           22.379167          26.033333
7/2/2023           18.608333           23.875000          29.279167

[697 rows x 3 columns]

In the table above, wind's average speed is faster than 14 km/h in 697 days out of 1280. This means that if wind turbines are built, the effectiveness and efficiency of wind turbines are about 54% which is risky. 
But the table is based on historical data and it comprises wind speed occurring in whole Uzbekistan, not in the parts of it. Analysing the future trends of wind and its speed in the parts of Uzbekistan may be beneficial to build wind turbines, because different regions have different relief.